File: c++/operadora.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {

    int minutos;
    double valorPago;

    cout << "Digite a quantidade de minutos: ";
    cin >> minutos;

    if (minutos > 100) {
        valorPago = ((minutos - 100) * 2) + 50.00;
        cout << fixed << setprecision(2);
        cout << "Valor a pagar: R$ " << valorPago << endl;
    } else {
        valorPago = 50.00;
        cout << fixed << setprecision(2);
        cout << "Valor a pagar: R$ " << valorPago << endl;
    }

    return 0;
}

File: java/operadora.java
import java.util.Locale;
import java.util.Scanner;

public class operadora {

	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		int minutos;
		double valor;
		
		System.out.print("Digite a quantidade de minutos: ");
		minutos = sc.nextInt();
		
		valor = 50.0;
		if (minutos < 100) {
			System.out.println("Valor a pagar: R$ " + String.format("%.2f", valor));
		} else {
			valor = valor + (minutos - 100) * 2.00;
			System.out.println("Valor a pagar: R$ " + String.format("%.2f", valor));
		}
		
		sc.close();
	}
	
}